dask/dataframe/core.py

```diff
@@ -2958,27 +2959,32 @@ def call(obj, attr, *args):
 
 def set_sorted_index(df, index, drop=True, **kwargs):
     if not isinstance(index, Series):
-        index2 = df[index]
         meta = df._meta.set_index(index, drop=drop)
     else:
-        index2 = index
         meta = df._meta.set_index(index._meta, drop=drop)
 
-    mins = index2.map_partitions(lambda x: pd.Series(x.min()), meta=index2)
-    maxes = index2.map_partitions(lambda x: pd.Series(x.max()), meta=index2)
+    result = map_partitions(_set_sorted_index, df, index, drop=drop, meta=meta)
+
+    return compute_divisions(result, **kwargs)
+
+
+def compute_divisions(df, **kwargs):
+    mins = df.index.map_partitions(lambda x: pd.Series(x.min()), meta=df.index)
+    maxes = df.index.map_partitions(lambda x: pd.Series(x.max()), meta=df.index)
```

Comment:
Won't `x[0]` and `x[-1]` be better than `x.min()` and `x.max()` here?

It's also a better test for invalid input IMHO